streams/src/test/java/org/apache/kafka/streams/processor/internals/RepartitionTopicsTest.java

```diff
@@ -0,0 +1,434 @@
+import static org.easymock.EasyMock.mock;
+import static org.easymock.EasyMock.niceMock;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Cluster.class})
+public class RepartitionTopicsTest {
+
+    private static final String SOURCE_TOPIC_NAME1 = "source1";
+    private static final String SOURCE_TOPIC_NAME2 = "source2";
+    private static final String SOURCE_TOPIC_NAME3 = "source3";
+    private static final String SINK_TOPIC_NAME1 = "sink1";
+    private static final String SINK_TOPIC_NAME2 = "sink2";
+    private static final String REPARTITION_TOPIC_NAME1 = "repartition1";
+    private static final String REPARTITION_TOPIC_NAME2 = "repartition2";
+    private static final String REPARTITION_TOPIC_NAME3 = "repartition3";
+    private static final String REPARTITION_TOPIC_NAME4 = "repartition4";
+    private static final String REPARTITION_WITHOUT_PARTITION_COUNT = "repartitionWithoutPartitionCount";
+    private static final String SOME_OTHER_TOPIC = "someOtherTopic";
+    private static final Map<String, String> TOPIC_CONFIG1 = Collections.singletonMap("config1", "val1");
+    private static final Map<String, String> TOPIC_CONFIG2 = Collections.singletonMap("config2", "val2");
+    private static final Map<String, String> TOPIC_CONFIG5 = Collections.singletonMap("config5", "val5");
+    private static final RepartitionTopicConfig REPARTITION_TOPIC_CONFIG1 =
+        new RepartitionTopicConfig(REPARTITION_TOPIC_NAME1, TOPIC_CONFIG1, 4, true);
+    private static final RepartitionTopicConfig REPARTITION_TOPIC_CONFIG2 =
+        new RepartitionTopicConfig(REPARTITION_TOPIC_NAME2, TOPIC_CONFIG2, 2, true);
+    private static final TopicsInfo TOPICS_INFO1 = new TopicsInfo(
+        mkSet(REPARTITION_TOPIC_NAME1),
+        mkSet(SOURCE_TOPIC_NAME1, SOURCE_TOPIC_NAME2),
+        mkMap(
+            mkEntry(REPARTITION_TOPIC_NAME1, REPARTITION_TOPIC_CONFIG1),
+            mkEntry(REPARTITION_TOPIC_NAME2, REPARTITION_TOPIC_CONFIG2)
+        ),
+        Collections.emptyMap()
+    );
+    private static final TopicsInfo TOPICS_INFO2 = new TopicsInfo(
+        mkSet(SINK_TOPIC_NAME1),
+        mkSet(REPARTITION_TOPIC_NAME1),
+        mkMap(mkEntry(REPARTITION_TOPIC_NAME1, REPARTITION_TOPIC_CONFIG1)),
+        Collections.emptyMap()
+    );
+
+    final InternalTopologyBuilder internalTopologyBuilder = mock(InternalTopologyBuilder.class);
+    final InternalTopicManager internalTopicManager = mock(InternalTopicManager.class);
+    final CopartitionedTopicsEnforcer copartitionedTopicsEnforcer = mock(CopartitionedTopicsEnforcer.class);
+    final Cluster clusterMetadata = niceMock(Cluster.class);
+
+    @Before
+    public void setUp() {
```

Comment:
Did you mean to leave in an empty method?

Reply:
No, I will remove it.